src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,84 @@
+"use client";
+
+import React, { ReactNode } from "react";
+import styled from "styled-components";
+import { Dialog } from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+
+const StyledDialog = styled(Dialog)`
+    & .MuiPaper-root {
+        border: 0;
+        border-radius: 2em;
+        padding: 0;
+        margin: 5%;
+        overflow: hidden;
+        max-width: 90%;
+    }
+`;
+
+const Header = styled.div`
+    display: flex;
+    position: sticky;
+    justify-content: space-between;
+    align-items: center;
+    color: ${(props) => props.theme.secondaryForegroundColor};
+    // TODO VFB-16 Change to using theme with palettes of colours
+    background-color: ${(props) => props.theme.secondaryBackgroundColor}A0;
+    font-size: 1.5em;
+    font-weight: bolder;
+    padding: 0.75em 1em;
+    width: 100%;
+`;
+
+const Content = styled.div`
+    overflow: auto;
+    padding: 1em 1.5em 1.5em;
+    width: 100%;
+`;
+
+const CloseButton = styled.button.attrs({
+    "aria-label": "Close Button",
+})`
+    border: 0;
+    border-radius: 50%;
+
+    // TODO VFB-16 Change to using theme with palettes of colours
+    color: grey;
+
+    display: flex;
+    justify-content: center;
+    aspect-ratio: 1;
+    align-items: center;
+    transition: 0.2s;
+    margin-left: 1em;
+
+    &:hover {
+        // TODO VFB-16 Change to using theme with palettes of colours
```

Comment:
As mentioned somewhere else, I don't want you to push hardcoded colours because 'it will be changed later' - main branch should be good to revert to at all times!
I'd much rather you use a themed colour (that might not be the same) and put in the comment exactly what you want this colour to look more like (e.g. `color: ...theme.foregroundColor // TODO VFB-16 Add 'grey' to palette of colours`)

Reply:
Refactored